lib/ramble/ramble/application.py

```diff
@@ -107,6 +108,42 @@ def __init__(self, file_path):
         self.close_logger()
         self.build_phase_order()
 
+        self._initialize_directives()
+
+    def _initialize_directives(self):
+        """Create class methods that execute directives
+
+        Wrap each directive, and inject it into this class instance as a class
+        method.
+
+        This allows:
+
+        self.<directive_name>(<directive_args>) to be called. As in:
+
+        self.archive_pattern('*.log')
+
+        Which can be called within `def __init__(self, file_path)` instead of
+        having to call `archive_pattern('*.log')` at the class definition level.
```

Comment:
No, you can call it elsewhere, but it wouldn't necessarily be configured in the right place.

i.e. you could call `self.figure_of_merit` in `_analyze_experiments` but you might not define it early enough to have it be processed correctly.

Reply:
Yea `_analyze_experiments` was the best example I could come up with. My gut was that the directive call would essentially do nothing since it has to be at init time, but I wanted to check

I wonder if we can warn somehow..